DOC-13648 document magma flusher threads 7.6 (#4079)
couchbase/docs-server: modules/learn/pages/buckets-memory-and-storage/storage-engines.adoc

[.edition]#{enterprise}#

Magma is designed for high performance with very large datasets that do not fit in memory.
It's ideal for use cases that rely primarily on disk access.
The performance of disk access will be as good as the underlying disk sub-systems --
for example, using NVMe SSDs will give higher performance.

Magma can work with very low amounts of memory for large datasets: a minimum memory-to-data ratio of 1% is required.
For example, if a node is holding 5{nbsp}TB of data, Magma can be used with only 64{nbsp}GB RAM.

Magma has several thread settings to tune performance for different workloads.
To get maximum performance from Magma for disk-oriented workloads, set the Writer Threads to `Disk i/o optimized`. This setting allocates more threads to sustain high write rates.

> See xref:manage:manage-settings/general-settings.adoc#data-settings[Data Settings] for more information about this setting.

Reviewer: ```suggestion
For more information about this setting, see xref:manage:manage-settings/general-settings.adoc#data-settings[Data Settings].
```

You can also tune the number and allocation of threads that Magma uses to compact and flush data to disk. 

> You may choose to change the ratio of threads that compact data to the threads that flush data if you notice that Magma's data compaction spikes CPU use.

Reviewer: ```suggestion
If you notice that Magma's data compaction spikes CPU use, you can change the ratio of threads that compact data to the threads that flush data.
```
